test/Microsoft.Azure.Devices.Edge.Test.Common/IotHub.cs

```diff
@@ -146,23 +146,31 @@ public Task<Device> CreateEdgeDeviceIdentityAsync(string deviceId, Authenticatio
             string deviceId,
             string moduleId,
             CloudToDeviceMethod method,
-            CancellationToken token)
+            CancellationToken token,
+            bool retry = true)
```

Comment:
This is a workaround for the problem your module has connecting to edge hub, right?

Reply:
It isn't a solution, it just makes it fail fast and actually show the error. For the test, if there's any kind of connection error, I don't want it to retry until timeout, I want it to fail.